### PR TITLE
Feature/lww 190 implement sticky asset details

### PR DESCRIPTION
**Pull request by andreyzhylin**

### CODE REVIEW

File: src/components/TransactionsTable/index.tsx [hunk 1]
```diff
@@ -239,17 +309,46 @@ export class TransactionsTable extends React.Component<TransactionsTableProps> {
...
   private handleLoadMoreTransactions = async () => {
+    if (this.areTransactionsLoading) {
+      return;
+    }
+
     this.pageNumber++;
     this.loadTransactions();
   };
+
+  private handleScroll = () => {
```
Comment (PaulRumyantsevItr) [inline src/components/TransactionsTable/index.tsx]: Have you tried to use ```position: sticky``` from css?

File: src/components/TransactionsTable/index.tsx [hunk 2]
```diff
@@ -239,17 +309,46 @@ export class TransactionsTable extends React.Component<TransactionsTableProps> {
   };
 
   private handleTransactionsFilterChange = async (
-    transactionType: TransactionType[]
+    transactionType: TransactionType[],
+    shouldScroll?: boolean
   ) => {
+    if (shouldScroll && this.filtersRowElement) {
+      const filtersRowStartPosition = this.filtersRowElement.offsetTop - 170;
```
Comment (PaulRumyantsevItr) [inline src/components/TransactionsTable/index.tsx]: Magic number. Move 170 to some constant.

File: src/components/TransactionsTable/index.tsx [hunk 3]
```diff
@@ -239,17 +309,46 @@ export class TransactionsTable extends React.Component<TransactionsTableProps> {
...
+    }
+
     this.pageNumber++;
     this.loadTransactions();
   };
+
+  private handleScroll = () => {
+    if (this.filtersRowElement) {
+      const stickyHeaderBreakpointPosition =
+        this.filtersRowElement.offsetTop - 30;
```
Comment (PaulRumyantsevItr) [inline src/components/TransactionsTable/index.tsx]: Same, magic number. Move to constant.

File: src/components/TransactionsTable/style.scss [hunk 4]
```diff
@@ -2,63 +2,56 @@
...
-    @media all and (max-width: $screen-tablet) {
-      @include rem(padding, 0);
-    }
   }
 
   .transactions-table {
     @include rem(margin-top, 12px);
     @include rem(padding, 0 94px);
 
+    &__asset-col,
```
Comment (GeorgePuisha) [inline src/components/TransactionsTable/style.scss]: Just `__column`

File: src/components/TransactionsTable/style.scss [hunk 5]
```diff
@@ -216,8 +209,58 @@
       text-align: center;
     }
 
+    .spinner_container {
+      margin-bottom: 2000px;
```
Comment (PaulRumyantsevItr) [inline src/components/TransactionsTable/style.scss]: Strange value for margin. Why are you using need it?

File: src/components/TransactionsTable/style.scss [hunk 6]
```diff
@@ -216,8 +209,58 @@
...
       @include rem(padding, 0 24px 0 0);
     }
   }
+
+  .sticky-row {
+    background: $white;
+    position: fixed;
+    top: -160px;
+    left: 0;
+    right: 0;
```
Comment (PaulRumyantsevItr) [inline src/components/TransactionsTable/style.scss]: One of the style (left or right) is redundant

File: src/components/TransactionsTable/style.scss [hunk 7]
```diff
@@ -216,8 +209,58 @@
...
+    margin: 0 auto;
+    z-index: 11111;
+    transition: all 0.5s;
+
+    &_active {
+      top: 0px;
```
Comment (PaulRumyantsevItr) [inline src/components/TransactionsTable/style.scss]: top: 0;

File: src/pages/AssetPage/index.tsx [hunk 8]
```diff
@@ -104,6 +104,16 @@ export class AssetPage extends React.Component<AssetPageProps> {
         <TransactionsTable
           transactions={this.transactionStore.assetTransactions}
           loadTransactions={this.loadTransactions}
+          stickyTitle={
+            <div className="sticky-title">
```
Comment (PaulRumyantsevItr) [inline src/pages/AssetPage/index.tsx]: I think it makes sense to move this title to separate component or, at least, return it using other render function

File: src/pages/HistoryPage/index.tsx [hunk 9]
```diff
@@ -38,6 +38,11 @@ export class HistoryPage extends React.Component<RootStoreProps> {
         <TransactionsTable
           transactions={this.transactionStore.walletTransactions}
           loadTransactions={this.loadTransactions}
+          stickyTitle={
+            <div className="sticky-title">
```
Comment (PaulRumyantsevItr) [inline src/pages/HistoryPage/index.tsx]: The same. Use other component or render function

File: src/pages/HistoryPage/style.scss [hunk 10]
```diff
@@ -22,3 +22,13 @@
     @include rem(font-size, 30px);
   }
 }
+
+.sticky-title {
```
Comment (GeorgePuisha) [inline src/pages/HistoryPage/style.scss]: `sticky-title` could be moved to the separate file (defined twice and could be reused if new pages will appear).

File: src/components/TransactionsTable/style.scss [hunk 11]
```diff
@@ -2,63 +2,56 @@
...
+      @include rem(font-size, 26px);
 
       @media all and (max-width: $screen-mobile-old) {
         display: block;
       }
     }
 
     &__item {
-      color: #636f7e;
+      color: #8c94a0;
```
Comment (GeorgePuisha) [inline src/components/TransactionsTable/style.scss]: Why not to make this a constant variable in `LykkeExchangeWeb/src/styles/common/_variables.scss`?

File: src/components/TransactionsTable/index.tsx (hunk 1 above)
Comment (andreyzhylin) [reply src/components/TransactionsTable/index.tsx]: Yeap, it's not applicaple in this case

File: src/components/TransactionsTable/style.scss (hunk 5 above)
Comment (andreyzhylin) [reply src/components/TransactionsTable/style.scss]: To make scroll persistent during loading

File: src/components/TransactionsTable/style.scss (hunk 6 above)
Comment (andreyzhylin) [reply src/components/TransactionsTable/style.scss]: Both are required to put this block in the middle https://www.w3.org/TR/css-position-3/#size-and-position-details

File: src/components/TransactionsTable/style.scss (hunk 4 above)
Comment (andreyzhylin) [reply src/components/TransactionsTable/style.scss]: `column` isn't element of block `asset`, `asset column` is element of block `transactions-table`, also element `asset` can't contain another element https://en.bem.info/methodology/quick-start/#element

**Pull request merged**